app/models/agents/telegram_agent.rb

```diff
@@ -79,7 +79,11 @@ def receive_event(event)
     def send_telegram_message(method, params)
       params[:chat_id] = interpolated['chat_id']
       params[:parse_mode] = interpolated['parse_mode'] if interpolated['parse_mode'].present?
-      HTTMultiParty.post telegram_bot_uri(method), query: params
+      txt_message = params[:text].scan /.{1,4096}\W/m
```

Comment:
You're right - my initial test ended with a non-letter. Yours looks close. I did a quick test of 

/\G(?:\w{4096}|.{1,4096}(?<=\s)|.{1,4096}(?=\b|\z))/m


`2.3.0 :001 > blah = "this is a test nothing other than that"`
 `=> "this is a test nothing other than that" `
`2.3.0 :002 > blah.length`
` => 38 `
`2.3.0 :003 > blah.scan(/\G(?:\w{38}|.{1,38}(?<=\s)|.{1,38}(?=\b|\z))/m)`
` => ["this is a test nothing other than ", "that"] `